### PR TITLE
Add maximum refinement level parameter.

### PR DESCRIPTION
**Pull request by jperryhouts**

### CODE REVIEW

File: source/simulator/parameters.cc [hunk 1]
```diff
@@ -489,6 +489,12 @@ namespace aspect
                          Patterns::Double(0,1),
                          "The fraction of cells with the smallest error that "
                          "should be flagged for coarsening.");
+      prm.declare_entry ("Maximum refinement level", "inf",
```
Comment (bangerth) [inline source/simulator/parameters.cc]: That's not a valid integer.

**Comment by bangerth** [comment]
I believe there is a plugin that limits the maximal mesh level. Why is this not enough?

**Comment by jperryhouts** [comment]
Yep. There's also a "Minimum refinement function" plugin, and a "Minimum refinement level" parameter. I was just going for consistency. Feel free to veto this feature, it is a bit redundant.

**Comment by tjhei** [comment]
So this is equivalent to
```
set Strategy = maximum refinement function
  subsection Maximum refinement function
    set Function expression    = 7
  end
end
```
right? I think it is not needed then.

**Comment by jperryhouts** [comment]
Yep, that's what I usually do. But so is

    set Minimum refinement level = 3

the same as

    set Strategy = Minimum refinement function
    subsection Minimum refinement function
      set Function expression = 3
    end

**Comment by bangerth** [comment]
Yes, the duplication of the minimum mesh refinement function is a historical accident :-(

**Pull request closed without merging**